### PR TITLE
Externalize (arms) dealer info

### PR DESCRIPTION
**Pull request by ja2-derek**
Externalizes (#665) `ARMS_DEALER_INFO`, which is the info for all bartenders/traders/repairmen in game. Dealer inventory are already externalized, and this PR externalizes the dealers themselves.

## Summary 

The first few commits are method extraction that should not change any functionalities.

The next commits are adding the JSON data, adding model classes, and replacing `ARMS_DEALER_INFO&` with `const DealerModel*`. New dealer flags are added but not yet used in this PR.

## What's next

This PR avoids any non-essential changes, as it already has many lines changed. I intend to make further PRs to generalize some hardcode logic (e.g. repairmen check, electronics repairer (Fredo) check, and so on), and maybe to relax the limit on number of dealers.

I will do a full round of testing after the follow-up PRs.

### CODE REVIEW

File: src/game/Tactical/ShopKeeper_Interface.cc [hunk 1]
```diff
@@ -1896,10 +1908,10 @@ static void DisplayArmsDealerCurrentInventoryPage(void)
 					fDisplayHatchOnItem = TRUE;
 				}
 				//if the dealer repairs
-				else if( ArmsDealerInfo[ gbSelectedArmsDealerID ].ubTypeOfArmsDealer == ARMS_DEALER_REPAIRS )
+				else if (DoesDealerDoRepairs(gbSelectedArmsDealerID))
 				{
 					//if the item is damaged
-					if( ArmsDealerInfo[ gbSelectedArmsDealerID ].uiFlags & ARMS_INV_ITEM_NOT_REPAIRED_YET )
+					if (gpTempDealersInventory[usCnt].uiFlags & ARMS_INV_ITEM_NOT_REPAIRED_YET)
```
Comment (ja2-derek) [inline src/game/Tactical/ShopKeeper_Interface.cc:1914]: This has to be a bug. The original code was looking for a `ARMS_INV_ITEM` flag in the `ARMS_DEALER_INFO`.

**Comment by ja2-derek** [comment]
Not sure why the tests is failing only on Mac. I am still looking for clues.

File: src/game/Tactical/ShopKeeper_Interface.cc (hunk 1 above)
Comment (lynxlynxlynx) [reply src/game/Tactical/ShopKeeper_Interface.cc:1914]: However the effect was that it was always false. Shouldn't this check be negated? As in, only hatch items that have been repaired already?

**Comment by lynxlynxlynx** [comment]
Other than the comment, LGTM! :+1:

File: src/game/Tactical/ShopKeeper_Interface.cc (hunk 1 above)
Comment (ja2-derek) [reply src/game/Tactical/ShopKeeper_Interface.cc:1914]: Indeed... the hatch seems totally useless. 
 1. The repaired item are placed in the bottom section
 2. The repair ETA cannot be read at all with hatch  

See the below screenshot:
image

I think we should just remove the check and leave `fDisplayHatchOnItem` as FALSE.

Comment (lynxlynxlynx) [reply src/game/Tactical/ShopKeeper_Interface.cc:1914]: Yes, let's remove — the red text is enough.

**Pull request merged**